api/application/client.go

```diff
@@ -102,6 +107,16 @@ type DeployArgs struct {
 // it. Placement directives, if provided, specify the machine on which the charm
 // is deployed.
 func (c *Client) Deploy(args DeployArgs) error {
+	if len(args.AttachStorage) > 0 && args.NumUnits != 1 {
+		return errors.Errorf("AttachStorage is non-empty, but NumUnits is %d", args.NumUnits)
```

Comment:
I think this could be more user friendly
"Cannot specify storage attachments when more than one unit is requested"
or something like that